src/mito/src/table.rs

```diff
@@ -322,6 +322,31 @@ impl<R: Region> Table for MitoTable<R> {
         }
         Ok(rows_deleted)
     }
+
+    async fn flush(&self, request: FlushTableRequest) -> TableResult<()> {
+        if let Some(region_id) = request.region_id {
+            if let Some(region) = self.regions.get(&region_id) {
+                region
+                    .flush()
+                    .await
+                    .map_err(BoxedError::new)
+                    .context(table_error::TableOperationSnafu)?;
+            };
+        } else {
+            let futs = self
+                .regions
+                .values()
+                .map(|region| region.flush())
+                .collect::<Vec<_>>();
+
+            futures::future::try_join_all(futs)
```

Comment:
[nitpick] you can spare an allocation
```suggestion
            futures::future::try_join_all(self.regions.values().map(R::flush))
```